galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java

```diff
@@ -46,295 +46,309 @@
-      Map<String, String> newParameterMap = new HashMap<>();
-      for (String parameterName : ParameterMap.keySet()) {
-         String[] values = ParameterMap.get(parameterName);
-         if (values != null && values.length > 0) {
-            newParameterMap.put(parameterName, values[0]);
-         } else {
-            newParameterMap.put(parameterName, null);
-         }
-      }
-   
-      return newParameterMap;
-   }
-  
-   
-   class SortByEndTime implements Comparator<RasRunResult> {
-      
-      @Override
-      public int compare(RasRunResult a, RasRunResult b) {
-         Instant aEndTime = a.getTestStructure().getEndTime();
-         Instant bEndTime = b.getTestStructure().getEndTime();
-         
-         if(aEndTime == null) {
-            if(bEndTime == null) {
-               return 0;
-            }
-            return -1;
-         }
-         
-         if(bEndTime == null) {
-            return 1;
-         }
-         
-         return aEndTime.compareTo(bEndTime);
-      }
-   }
-   
-   class SortByTestClass implements Comparator<RasRunResult>{
-     
-      @Override
-      public int compare(RasRunResult a, RasRunResult b) {
-        String aTestClass = a.getTestStructure().getTestShortName();
-        String bTestClass = b.getTestStructure().getTestShortName();
-        
-        if(aTestClass == null) {
-           if(bTestClass == null) {
-              return 0;
-        }
-           return -1;
-      }
-       if(bTestClass == null) {
-          return 1;
-       }
-       
-       return aTestClass.compareTo(bTestClass);
-   }
-      
- }
+	@Reference
+	IFramework framework;
+
+	private static final long serialVersionUID = 1L;
+
+	final static Gson gson = GalasaGsonBuilder.build();
+
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+		int pageNum = 1;
+		int pageSize = 100;
+
+		Map<String,String> paramMap = getParameterMap(req);
+
+		if (paramMap.get("page") != null && !paramMap.get("page").equals("")) {
+			try {
+				pageNum = Integer.parseInt(paramMap.get("page"));
+			} catch (Exception e) {
+				throw new ServletException("Error parsing integer, ",e);
+			}
+		}
+
+		if (paramMap.get("size") != null && !paramMap.get("size").equals("")) {
+			try {
+				pageSize = Integer.parseInt(paramMap.get("size"));
+			} catch (Exception e) {
+				throw new ServletException("Error parsing integer, ", e);
+			}
+		}
+
+		List<RasRunResult> runs = new ArrayList<>();
+
+		/* Get list of Run Ids from the URL -
+		If a Run ID parameterlist is present in the URL then only return that run / those runs
+		Do not filter as well */
+
+		String runIdsParam = "";
+		if (paramMap.get("runId") != null && !paramMap.get("runId").isEmpty()) {
+			runIdsParam = paramMap.get("runId");
+			
+			String [] runIds = runIdsParam.split("[,]");
+			for (String runId : runIds) {
+				try {
+					IRunResult run = getRunByRunId(runId);
```

Comment:
what happens if the runid is not found?

Reply:
ie,  if the runid is invalid or deleted,  it should be ignored and not added to the array